gredux_test.go

```diff
@@ -153,16 +153,40 @@ func BenchmarkDispatch(b *testing.B) {
 		count int
 	}
 	store := New(counterState{0})
-	store.Reducer(func(state State, action Action) State {
+	store.Reducer(func(state State, action Action) (State, interface{}) {
 		switch action.ID {
 		case "increment":
-			return counterState{state.(counterState).count + 1}
+			return counterState{state.(counterState).count + 1}, nil
 		default:
-			return state
+			return state, nil
 		}
 	})
 
 	for i := 0; i < b.N; i++ {
 		store.Dispatch(Action{"increment", nil})
 	}
 }
+
+func TestConcurrentSelectorSupport(t *testing.T) {
+	type testState struct {
+		success bool
+	}
+	store := New(testState{false})
+	store.Reducer(func(state State, action Action) (State, interface{}) {
+		switch action.ID {
+		case "select":
+			return state, state.(testState).success
+		default:
+			return testState{true}, nil
+    }
+	})
+	for i := 0; i < 10; i++ {
+		go func() {
```

Comment:
This test will exit immediately without waiting for these goroutines. You should create a channel or a sync/waitgroup to wait for all of these goroutines to finish. I don't think it's really necessary to test this functionality concurrently since there are already tests which verify that dispatch works concurrently without race conditions.